tommath.h

```diff
@@ -51,37 +55,35 @@ extern "C" {
  * At the very least a mp_digit must be able to hold 7 bits
  * [any size beyond that is ok provided it doesn't overflow the data type]
  */
+
 #ifdef MP_8BIT
 typedef uint8_t              mp_digit;
 typedef uint16_t             private_mp_word;
-#   define MP_SIZEOF_MP_DIGIT 1
-#   ifdef MP_DIGIT_BIT
-#      error You must not define MP_DIGIT_BIT when using MP_8BIT
-#   endif
+#   define MP_DIGIT_BIT 7
 #elif defined(MP_16BIT)
 typedef uint16_t             mp_digit;
 typedef uint32_t             private_mp_word;
-#   define MP_SIZEOF_MP_DIGIT 2
-#   ifdef MP_DIGIT_BIT
-#      error You must not define MP_DIGIT_BIT when using MP_16BIT
-#   endif
+#   define MP_DIGIT_BIT 15
 #elif defined(MP_64BIT)
 /* for GCC only on supported platforms */
 typedef uint64_t mp_digit;
 typedef unsigned long        private_mp_word __attribute__((mode(TI)));
 #   define MP_DIGIT_BIT 60
 #else
-/* this is the default case, 28-bit digits */
-
-/* this is to make porting into LibTomCrypt easier :-) */
 typedef uint32_t             mp_digit;
 typedef uint64_t             private_mp_word;
-
 #   ifdef MP_31BIT
-/* this is an extension that uses 31-bit digits */
+/*
+ * This is an extension that uses 31-bit digits.
+ * Please be aware that not all functions support this size, especially s_mp_mul_digs_fast
+ * will be reduced to work on small numbers only:
+ * Up to 8 limbs, 248 bits instead of up to 512 limbs, 15872 bits with MP_28BIT.
+ */
 #      define MP_DIGIT_BIT 31
 #   else
-/* default case is 28-bit digits, defines MP_28BIT as a handy macro to test */
+/* default case is 28-bit digits, defines MP_28BIT as a handy macro to test
+ * this is to make porting into LibTomCrypt easier :-)
```

Comment:
this comment isn't valid anymore and can be removed :) 
The last usage of `MP_28BIT` disappeared when ltc was separated into ltc and ltm

Reply:
I will fix this and rebase the other open PRs I mentioned as ready.

Reply:
yes please fix it
I'll rebase one after another after merging